File: linux/ardium_source/presentation/07_JSON/json.swift
// Topic: JSON Handling
// Parse JSON string

import Foundation

struct Config: Codable {
    let name: String
    let vers: Int
}

let jsonStr = "{\"name\": \"Ardium\", \"vers\": 2}"
let jsonData = jsonStr.data(using: .utf8)!

let decoder = JSONDecoder()
if let config = try? decoder.decode(Config.self, from: jsonData) {
    print("Name: \(config.name)")
    print("Version: \(config.vers)")
}

File: linux/ardium_source/presentation/07_JSON/json.rs
// Topic: JSON Handling (Mock)
// Parse: {"name": "Ardium", "vers": 2}

struct Config {
    name: String,
    vers: i32,
}

fn main() {
    let json = "{\"name\": \"Ardium\", \"vers\": 2}";
    
    // Mock parsing for demo without external dependencies
    // In real Rust, use serde_json
    
    let name = "Ardium";
    let vers = 2;
    
    println!("Name: {}", name);
    println!("Version: {}", vers);
}
